src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionSecurityTokenAuthenticator.cs

```diff
@@ -0,0 +1,1366 @@
+                {
+                    this.issuedSecurityTokenHandler(newSessionToken, remoteAddress);
+                }
+                else
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new SecurityNegotiationException(SR.IssueSessionTokenHandlerNotSet));
+                }
+            }
+            else if (operation == SecuritySessionOperation.Renew)
+            {
+                if (this.renewedSecurityTokenHandler != null)
+                {
+                    this.renewedSecurityTokenHandler(newSessionToken, previousSessionToken);
+                }
+                else
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperWarning(new SecurityNegotiationException(SR.RenewSessionTokenHandlerNotSet));
+                }
+            }
+            else
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new NotSupportedException());
+            }
+        }
+
+        public override Task CloseAsync(CancellationToken token)
+        {
+           /* TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            if (this.rstListener != null)
+            {
+                this.rstListener.Close(timeoutHelper.RemainingTime());
+                this.rstListener = null;
+            }*/
+           
+           return base.CloseAsync(token);
+        }
+
+        public override Task OpenAsync(CancellationToken token)
+        {
+            if (this.BootstrapSecurityBindingElement == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.BootstrapSecurityBindingElementNotSet, this.GetType())));
+            }
+            if (this.IssuerBindingContext == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.IssuerBuildContextNotSet, this.GetType())));
+            }
+            if (this.IssuedSecurityTokenParameters == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.IssuedSecurityTokenParametersNotSet, this.GetType())));
+            }
+            if (this.SecurityAlgorithmSuite == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.SecurityAlgorithmSuiteNotSet, this.GetType())));
+            }
+            if (this.IssuedTokenCache == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.IssuedTokenCacheNotSet, this.GetType())));
+            }
+            //TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+            SetupSessionListener();
+           // this.rstListener.Open(timeoutHelper.RemainingTime());
```

Comment:
Should be calling open on the session listener. As things have been refactored and there isn't an open method any more, just an InitializeRuntime method, can you see if you can do the initialization here? I'm a little nervous about changing the order that things execute.

Reply:
Matt, understand your concern... please take a look at OnOpenAsync(TimeSpan timeout) - SecuritySessionServerSettings... the old way and new way remain almost same.. except no open method for SecuritySessionHost.. the InitializeRuntime invoked from the above OpenAsync... let me know your thought...